src/hotspot/share/gc/shenandoah/shenandoahUnload.cpp

```diff
@@ -51,7 +51,8 @@ class ShenandoahIsUnloadingOopClosure : public OopClosure {
 
 public:
   ShenandoahIsUnloadingOopClosure() :
-    _marking_context(ShenandoahHeap::heap()->complete_marking_context()),
+    // TODO: In non-generational mode, this should still be complete_marking_context()
+    _marking_context(ShenandoahHeap::heap()->marking_context()),
```

Comment:
Is this not causing troubles in non-generational mode?

Reply:
`complete_marking_context` just asserts that marking is complete before returning the mark context. I'll revert this and test.

Reply:
https://bugs.openjdk.org/browse/JDK-8339127